specification/cognitiveservices/data-plane/Language/preview/2021-07-15-preview/questionanswering.json

```diff
@@ -346,25 +347,43 @@
       "additionalProperties": false,
       "properties": {
         "metadata": {
-          "type": "object",
-          "additionalProperties": {
-            "type": "string",
-            "minLength": 0,
-            "maxLength": 500
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MetadataRecord"
           }
         },
-        "compoundOperation": {
+        "logicalOperation": {
           "type": "string",
-          "$ref": "#/definitions/CompoundOperationKind",
+          "description": "Operation used to join metadata filters.",
+          "$ref": "#/definitions/LogicalOperationKind",
           "default": "AND"
         }
       }
     },
-    "CompoundOperationKind": {
+    "MetadataRecord": {
```

Comment:
Why are you redefining what a JSON object already is? JSON objects are already dictionaries. In the SDK we wouldn't expose an array of key/value pairs because that's what a dictionary is arleady, and takes care of making sure keys are unique unlike an array.

So why go back? How will the service handle the same key name in an array? Does first win, or last? Or do you err?

We'll also have to do work in the SDK to make this look like a dictionary either way, which those issues in the sentence directly above will affect as well.

Reply:
/cc @JeffreyRichter @johanste

Reply:
@heaths - The problem with using a dictionary is that we cannot support duplicate keys.

With the support of 'logicalOperation' we want to allow clients add multiple filters with same string in the key section. Our backend will handle the logic of filtering QnAs as mentioned in the request.

For e.g.,

`"metadataFilter": {
  "metadata": [
            {
              "key": "category",
              "value": "api1"
            },
            {
              "key": "category",
              "value": "api2"
            }
          ],
          "logicalOperation": "OR"
  }`

Is there a better way to implement it using something like Tuples?

Reply:
I see. Is it too late to use `$filter` query param like OData rather than inventing new filtering syntax?

Reply:
As discussed offline, which check on `$filter` but the assumption is that it's too late. For now, though, resolving.